lib/serve/functions.js

```diff
@@ -1,158 +1,14 @@
-
-  controller = new EmulatorController(controllerConfig);
-  var firebaseConfig;
-
-  ensureDefaultCredentials();
-  return functionsConfig.getFirebaseConfig(projectId, options.instance)
-  .then(function(result) {
-    firebaseConfig = JSON.stringify(result);
-    process.env.FIREBASE_PROJECT = firebaseConfig;
-    return controller.start();
-  }).then(function() {
-    return parseTriggers(projectId, functionsDir, firebaseConfig);
-  }).catch(function(e) {
-    utils.logWarning(chalk.yellow('functions:') + ' Failed to load functions source code. ' +
-      'Ensure that you have the latest SDK by running ' + chalk.bold('npm i --save firebase-functions') +
-      ' inside the functions directory. Please note that emulation of custom config values are not supported yet. ' +
-      'Run ' + chalk.bold('firebase serve --only hosting') + ' to only serve hosting files.\n\n' + e);
-    return RSVP.resolve();
-  }).then(function(triggers) {
-    var functionsEmulated = 0;
-    var promises = _.map(triggers, function(trigger) {
-      if (trigger.httpsTrigger) {
-        functionsEmulated += 1;
-        return controller.deploy(trigger.name, {
-          localPath: functionsDir,
-          triggerHttp: true
-        }).catch(function() {
-          return RSVP.reject({name: trigger.name});
-        });
-      }
-      return RSVP.resolve(); // TODO: support other trigger types
-    });
-    if (functionsEmulated > 0) {
-      track('Functions Emulation', 'HTTPS', functionsEmulated);
-    }
-    return RSVP.allSettled(promises);
-  }).then(function(operations) {
-    var functionUrls = {};
-    return RSVP.all(_.map(operations, function(operation) {
-      if (operation.state === 'rejected') {
-        utils.logWarning(chalk.yellow('functions:') + ' Failed to emulate ' + operation.reason.name);
-        return RSVP.resolve();
-      }
-      if (!operation.value) {
-        return RSVP.resolve(); // it was a non-HTTPS trigger
-      }
-      return _pollOperation(operation.value).then(function(res) {
-        var funcName = _.chain(res).get('name').split('/').last().value();
-        functionUrls[funcName] = res.httpsTrigger.url;
-      });
-    })).then(function() {
-      return functionUrls;
-    });
-  }).then(function(functionUrls) {
-    if (_.size(functionUrls) === 0) {
-      utils.logBullet(chalk.cyan.bold('functions: ') + 'No HTTPS functions emulated. ' +
-        'Support for other function types are coming soon.');
-      return;
-    }
-    _.forEach(functionUrls, function(url, funcName) {
-      utils.logSuccess(chalk.green.bold('functions: ') + funcName + ': ' + chalk.bold(url));
-    });
-  }).catch(function(e) {
-    utils.logWarning(chalk.yellow('functions:') + ' Error from emulator. ' + e.stack);
-    return _stopEmulator();
-  });
-}
+var FunctionsEmulator = require('../functionsEmulator');
+var emulatorInstance;
 
 module.exports = {
-  start: _startEmulator,
-  stop: _stopEmulator
+  start: function(options) {
```

Comment:
I've made both serve/functions.js and serve/hosting.js modules with "start" and "stop". In the future, when we add more targets for serving, this consistent file structure will make serve/index.js cleaner (similar to how deploy/index.js is set up). That's why I still have this file, instead of just importing FunctionsEmulator directly in serve/index.js